### DISCUSSION TITLE
Add support for security system devices

### DISCUSSION THREAD

**Discussion by sumnerboy12** [andrei-tatar/node-red-contrib-smartnora#69] [Ideas]
https://developers.google.com/assistant/smarthome/guides/securitysystem

This would allow integration of a house alarm and using Google Assistant to "arm house alarm" etc.

**Comment by andrei-tatar** [comment]
The service already supports the trait and the device type.
It was contributed by [Sebastian](https://github.com/sseidelmann) in this PR: https://github.com/andrei-tatar/nora-firebase-common/pull/3

The only left thing is to add the node-red node and update the doc.

**Reply by sumnerboy12** [reply]
Brilliant - sorry didn't realise it was already in-progress. Only just found out about the wonders of Smart NORA - thanks for an awesome bit of software!

**Comment by andrei-tatar** [comment]
I guess the question is do we switch to an async flow for this one, being security critical?

Meaning, an `arm` command that comes from google outputs an `arm` command on the node output but doesn't immediately update the state.

**Reply by sumnerboy12** [reply]
I like the sound of this - I would much prefer state to be updated by the device itself, rather than assuming just because a command has been sent.

**Reply by sumnerboy12** [reply]
Is there a reason all nodes are not async, or at least provide an option for them to be? I would like my `lock` node to be async, and only have the state update on a successful action, based on feedback from the lock itself.

**Reply by andrei-tatar** [reply]
No good reason, mostly for simplicity.

**Reply by andrei-tatar** [reply]
Also there are 2 different ways going about async commands:

1. when a command is received, the node outputs the command, node-red handles it and it should pipe back into the node as a command response. All this time the command handler should be blocked. The problem is Google recommends the execution to take under 1 sec but this is now out of NORA's control (I mean NORA could timeout after 1 sec with an error code - device not responding or something similar).

2. Use [follow-up responses](https://developers.google.com/assistant/smarthome/develop/notifications#events). Meaning a command is outputed but now the handler is not blocked anymore and the device can come later with a follow-up response. The problem here is only a few traits support follow-up responses (LockUnlock, OpenClose).

**Reply by sumnerboy12** [reply]
Yeah that is a real shame more don't support the follow-up responses. Seems like the "right" and most accurate way to do things, even if only optionally (i.e. if you have the feedback loop in place).

**Comment by andrei-tatar** [comment]
@sumnerboy12 I released version `v1.3.0-beta.0` of `node-red-contrib-smartnora` which includes a 1st version for the security system with ArmDisarm trait. 
Let me know if you can have a look and provide any feedback.

**Reply by sumnerboy12** [reply]
Seems to be working well, thanks very much!
